pkg/client/connector/install_test.go

```diff
@@ -509,21 +509,150 @@ func TestAddAgentToReplicaSet(t *testing.T) {
 			if !assert.NoError(t, actualErr) {
 				return
 			}
-			sanitizeDeployment(actualDep)
+			sanitizeWorkload(actualDep)
 
 			actualErr = undoServiceMods(ctx, actualSvc)
 			if !assert.NoError(t, actualErr) {
 				return
 			}
-			sanitizeDeployment(actualDep)
+			sanitizeWorkload(actualDep)
 
 			assert.Equal(t, expectedDep, actualDep)
 			assert.Equal(t, expectedSvc, actualSvc)
 		})
 	}
 }
 
-func sanitizeDeployment(obj kates.Object) {
+// I (Donny) would like to unify this w/ the "TestAddAgentToWorkload
```

Comment:
I know this is gross, the thing tripping me up on this is the DeepCopy doesn't exist for kates.Objects, so I can't replace kates.StatefulSet/ReplicaSet/Deployment with kates.Object.  I do want to fix these three tests to be less copy paste and remove redundancies, but I didn't think it should hold up getting the feature out.

Reply:
`kates.Object` implements `proto.Message`, right? You should be able to use `proto.Clone(obj)` on it for a slightly-less-efficient `DeepCopy`.